[Tuya] Not all presumed dimmer channels are dimmer channel

After upgrading from openHAB 5.2.0.Snapshot april 2026 to 5.2.0.M5 my Wifi DIN switch is no longer recognized well.

Below is the trace log of adding the DIN switch to openHAB.
The device ID is bf0acfb055d6b19164by7k.  Of course there is also info from other devices.

This is the line describing the error 👍 

- 2026-06-06 22:20:11.937 [ERROR] [nternal.common.AbstractInvocationHandler] - An error occurred while calling method 'ThingHandler.initialize()' on 'org.openhab.binding.tuya.internal.handler.TuyaDeviceHandler@513221df': Channel type 'tuya:rh8slxo4otrhaqx8_temp_value' is not known
- java.lang.IllegalArgumentException: Channel type 'tuya:rh8slxo4otrhaqx8_temp_value' is not known

The issue is about the temp_value. In the Tuya Platform it is defined as an Integer but openHAB reads it (from the tracelog) as an String I think.


At startup this error appears:
`09:50:05.981 Channel temp_value creation aborted, unit  "°C" has no known dimension, please report as bug.`

It would be nice if this could be repaired. When more/other information is needed, please tell me.

[Tuyalog.txt](https://github.com/user-attachments/files/28671948/Tuyalog.txt)

--- a/bundles/org.openhab.binding.tuya/src/main/java/org/openhab/binding/tuya/internal/TuyaChannelTypeProvider.java
+++ b/bundles/org.openhab.binding.tuya/src/main/java/org/openhab/binding/tuya/internal/TuyaChannelTypeProvider.java
@@ -256,13 +256,7 @@ public class TuyaChannelTypeProvider implements ChannelTypeProvider {
         StateDescriptionFragmentBuilder stateDescriptionFragmentBuilder = null;
         boolean advanced = false;
 
-        if (DIMMER_CHANNEL_CODES.contains(channelTypeId)) {
-            acceptedItemType = DIMMER;
-            category = "slider";
-            configurationRef = "channel-type:tuya:dimmer";
-            tags.add(schemaDp.readOnly ? "Status" : "Control");
-            tags.add("Brightness");
-        } else if ("bitmap".equals(schemaDp.type)) {
+        if ("bitmap".equals(schemaDp.type)) {
             acceptedItemType = NUMBER;
             category = "";
             configurationRef = "channel-type:tuya:bitmap";
@@ -305,7 +299,14 @@ public class TuyaChannelTypeProvider implements ChannelTypeProvider {
             category = "";
             configurationRef = "channel-type:tuya:number";
 
-            if (!schemaDp.unit.isEmpty()) {
+            if ((schemaDp.unit.isEmpty() && DIMMER_CHANNEL_CODES.contains(channelTypeId)) //
+                    || (!schemaDp.readOnly && "%".equals(schemaDp.unit))) {
+                acceptedItemType = DIMMER;
+                category = "slider";
+                configurationRef = "channel-type:tuya:dimmer";
+                tags.add(schemaDp.readOnly ? "Status" : "Control");
+                tags.add("Brightness");
+            } else if (!schemaDp.unit.isEmpty()) {
                 Unit<?> unit = schemaDp.parsedUnit;
                 if (unit == null) {
                     unit = UnitUtils.parseUnit(schemaDp.unit);

--- a/bundles/org.openhab.binding.tuya/src/main/java/org/openhab/binding/tuya/internal/handler/TuyaDeviceHandler.java
+++ b/bundles/org.openhab.binding.tuya/src/main/java/org/openhab/binding/tuya/internal/handler/TuyaDeviceHandler.java
@@ -23,7 +23,6 @@ import static org.openhab.binding.tuya.internal.TuyaBindingConstants.CONFIG_MIN;
 import static org.openhab.binding.tuya.internal.TuyaBindingConstants.CONFIG_PRODUCT_ID;
 import static org.openhab.binding.tuya.internal.TuyaBindingConstants.CONFIG_PROTOCOL;
 import static org.openhab.binding.tuya.internal.TuyaBindingConstants.CONFIG_RANGE;
-import static org.openhab.binding.tuya.internal.TuyaBindingConstants.DIMMER_CHANNEL_CODES;
 import static org.openhab.core.library.CoreItemFactory.COLOR;
 import static org.openhab.core.library.CoreItemFactory.DIMMER;
 import static org.openhab.core.library.CoreItemFactory.NUMBER;
@@ -645,10 +644,7 @@ public class TuyaDeviceHandler extends BaseThingHandler implements DeviceInfoSub
             Map<@Nullable String, @Nullable Object> configuration = new HashMap<>();
             configuration.put(CONFIG_DP, schemaDp.id);
 
-            if (DIMMER_CHANNEL_CODES.contains(channelId)) {
-                configuration.put(CONFIG_MIN, schemaDp.min);
-                configuration.put(CONFIG_MAX, schemaDp.max);
-            } else if ("enum".equals(schemaDp.type)) {
+            if ("enum".equals(schemaDp.type)) {
                 List<String> range = Objects.requireNonNullElse(schemaDp.range, List.of());
                 configuration.put(CONFIG_RANGE, String.join(",", range));
             } else if ("value".equals(schemaDp.type)) {

